nodes/analyzer/component_analyzer.py

```diff
@@ -0,0 +1,359 @@
+}
+def split_output(result_vals):
+    result_vals = [[[v] for v in r] for r in result_vals]
+    result_vals = lists_flat([result_vals])[0]
+    return lists_flat([result_vals])[0]
+
+class SvComponentAnalyzerNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Center/Matrix/Length
+    Tooltip: Data from vertices/edges/faces as Orientation, Location, Length, Normal, Center...
+
+    """
+
+    bl_idname = 'SvComponentAnalyzerNode'
+    bl_label = 'Component Analyzer'
+    bl_icon = 'VIEWZOOM'
+
+
+    modes = [
+        ('Verts', "Vertices", "Vertices Operators", 0),
+        ('Edges', "Edges", "Edges Operators", 1),
+        ('Faces', "Faces", "Faces Operators", 2)
+    ]
+
+    edge_modes =        [(k, k, descr, ident) for k, (ident, _, _, _, _, _, _, descr) in sorted(edges_modes_dict.items(),  key=lambda k: k[1][0])]
+    vertex_modes =      [(k, k, descr, ident) for k, (ident, _, _, _, _, _, _, descr) in sorted(vertex_modes_dict.items(), key=lambda k: k[1][0])]
+    face_modes =        [(k, k, descr, ident) for k, (ident, _, _, _, _, _, _, descr) in sorted(faces_modes_dict.items(),  key=lambda k: k[1][0])]
+    pols_origin_modes = [(k, k, descr, ident) for k, (ident, _, descr) in sorted(pols_origin_modes_dict.items(), key=lambda k: k[1][0])]
+    tangent_modes =     [(k, k, descr, ident) for k, (ident, _, descr) in sorted(tangent_modes_dict.items(), key=lambda k: k[1][0])]
+
+    origin_modes = [
+        ("Center", "Center", "Median Center", 0),
+        ("First", "First", "First Vertex", 1),
+        ("Last", "Last", "Last Vertex", 2)
+    ]
+    matrix_track_modes = [
+        ("X", "X", "Aligned with X", 0),
+        ("Y", "Y", "Aligned with Y", 1),
+        ("Z", "Z", "Aligned with Z", 2),
+        ("-X", "-X", "Aligned with -X", 3),
+        ("-Y", "-Y", "Aligned with -Y", 4),
+        ("-Z", "-Z", "Aligned with -Z", 5)
+    ]
+    matrix_normal_modes = [
+        ("X", "X", "Aligned with X", 0),
+        ("Z", "Z", "Aligned with Z", 2),
+    ]
+
+
+    @throttled
+    def update_mode(self, context):
+        # for mode in self.modes:
+        info = modes_dicts[self.mode][self.actual_mode()]
+
+        input_names = info[1]
+        output_socket_type = info[5]
+        output_socket_name = info[6].split(', ')
+        # hide unnecesary inputs only if not connected
+        for input_socket, key_name in zip(self.inputs, 'vep'):
+            if not input_socket.is_linked:
+                if not key_name in input_names:
+                    input_socket.hide_safe = True
+                else:
+                    input_socket.hide_safe = False
+
+        for idx, s in enumerate(output_socket_type):
+            self.outputs[idx].name = output_socket_name[idx]
+            self.outputs[idx].replace_socket(socket_dict[s])
+        if len(output_socket_type) < len(self.outputs):
+            for s in self.outputs[len(output_socket_type):]:
+                s.hide_safe = True
+        updateNode(self, context)
```

Comment:
`updateNode(self, context)` can be dropped, the `@throttled` decorator will call it as soon as the function ends.

https://github.com/nortikin/sverchok/blob/6896fb76c6c8c05d93e586a851d07488b00e6405/node_tree.py#L108-L134